plugins/k8sauditeks/pkg/k8sauditeks/k8sauditeks.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/falcosecurity/plugins/shared-aws/shared/go/aws/cloudwatchlogs"
+	"github.com/falcosecurity/plugins/shared-aws/shared/go/aws/session"
+)
+
+const pluginName = "k8sauditeks"
+
+type Plugin struct {
+	k8saudit.Plugin
+	Config PluginConfig
+}
+
+type PluginConfig struct {
+	Region          string `json:"region"           jsonschema:"description=The Region of your EKS cluster (default: us-east-1), it can be replaced by env var AWS_REGION"`
+	Profile         string `json:"profile"          jsonschema:"description=The Profile to use to create the session, it can be replaced by env var AWS_PROFILE"`
+	ClusterName     string `json:"cluster_name"     jsonschema:"description=The name of your EKS Cluster,required"`
+	UseAsync        bool   `json:"use_async"        jsonschema:"description=If true then async extraction optimization is enabled (Default: true)"`
+	PollingInterval uint64 `json:"polling_interval" jsonschema:"description=Polling Interval in seconds (default: 5s)"`
+	Shift           uint64 `json:"shift"            jsonschema:"description=Time shift in past in seconds (default: 1s)"`
+	BufferSize      uint64 `json:"buffer_size"      jsonschema:"description=Buffer Size (default: 200)"`
+}
+
+func (k *Plugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:          9,
+		Name:        pluginName,
+		Description: "Read Kubernetes Audit Events for EKS from Cloudwatch Logs",
+		Contact:     "github.com/falcosecurity/plugins",
+		Version:     "0.1.0",
+		EventSource: "k8s_audit",
+	}
+}
+
+// setDefault sets the configuration to its default values
+func (k *PluginConfig) setDefault() {
+	if p := os.Getenv("AWS_DEFAULT_PROFILE"); p != "" {
+		k.Profile = p
+	}
+	if p := os.Getenv("AWS_PROFILE"); p != "" {
+		k.Profile = p
+	}
+	if r := os.Getenv("AWS_DEFAULT_REGION"); r != "" {
+		k.Region = r
+	}
+	if r := os.Getenv("AWS_REGION"); r != "" {
+		k.Region = r
+	}
+	if k.Region == "" {
+		k.Region = "us-east-1"
+	}
+	k.UseAsync = true
+	// for PollingInterval, Shift and BufferSize, the default values from the package are used automatically
+}
+
+func (k *Plugin) Init(cfg string) error {
+	// read configuration
+	k.Config.setDefault()
+	err := json.Unmarshal([]byte(cfg), &k.Config)
+	if err != nil {
+		return err
+	}
+
+	// setup optional async extraction optimization
+	extract.SetAsync(k.Config.UseAsync)
+
+	return nil
+}
+
+func (p *Plugin) InitSchema() *sdk.SchemaInfo {
+	reflector := jsonschema.Reflector{
+		// all properties are optional by default
+		RequiredFromJSONSchemaTags: true,
+		// unrecognized properties don't cause a parsing failures
+		AllowAdditionalProperties: true,
+	}
+	if schema, err := reflector.Reflect(&PluginConfig{}).MarshalJSON(); err == nil {
+		return &sdk.SchemaInfo{
+			Schema: string(schema),
+		}
+	}
+	return nil
+}
+
+func (k *Plugin) Open(params string) (source.Instance, error) {
+	filter := cloudwatchlogs.CreateFilter("", "/aws/eks/"+k.Config.ClusterName+"/cluster", "kube-apiserver-audit", nil)
```

Comment:
Maybe `kube-apiserver-audit` should be configurable too.

Reply:
It's a string enforced by AWS for the logs.